src/emailservice/email_server.py

```diff
@@ -28,39 +28,13 @@
 from grpc_health.v1 import health_pb2
 from grpc_health.v1 import health_pb2_grpc
 
-from opentelemetry import trace
-from opentelemetry import propagators
-from opentelemetry.sdk.trace import TracerProvider
-from opentelemetry.exporter import zipkin
-from opentelemetry.sdk.trace.export import BatchExportSpanProcessor
-from opentelemetry.sdk.trace.propagation.b3_format import B3Format
-from opentelemetry.instrumentation.grpc import GrpcInstrumentorServer
-from opentelemetry.instrumentation.grpc import GrpcInstrumentorClient
-from opentelemetry.instrumentation.grpc.grpcext import intercept_server
+from opentelemetry.instrumentation.grpc import server_interceptor
```

Comment:
I don't think you should need this as `start_tracing()` will auto instrument the service.

Reply:
ack, tested and updated! ty!